v0/destinations/klaviyo/util.js

```diff
@@ -0,0 +1,175 @@
+const get = require("get-value");
+const { httpGET } = require("../../../adapters/network");
+const logger = require("../../../logger");
+
+const { WhiteListedTraits } = require("../../../constants");
+
+const {
+  defaultRequestConfig,
+  constructPayload,
+  getFieldValueFromMessage,
+  isDefinedAndNotNull,
+  removeUndefinedValues,
+  defaultPostRequestConfig,
+  extractCustomFields,
+  removeUndefinedAndNullValues
+} = require("../../util");
+
+const {
+  BASE_ENDPOINT,
+  LIST_CONF,
+  MAPPING_CONFIG,
+  CONFIG_CATEGORIES
+} = require("./config");
+
+/**
+ * This function is used to check if the user/profile already exists or not, if already exists unique person_id 
+ * for that user is getting returned else false is returned.
+ * @param {*} message 
+ * @param {*} Config 
+ * @returns
+ */
+const isProfileExist = async (message, { Config }) => {
+  const { privateApiKey } = Config;
+  const userIdentifiers = {
+    email: getFieldValueFromMessage(message, "email"),
+    external_id: getFieldValueFromMessage(message, "userId"),
+    phone_number: getFieldValueFromMessage(message, "phone")
+  };
+  //   const identifiers = Object.keys(userIdentifiers);
+  let personId;
+  for (const id in userIdentifiers) {
+    if (isDefinedAndNotNull(userIdentifiers[id]) && !personId) {
+      const profileResponse = await httpGET(
+        `${BASE_ENDPOINT}/api/v2/people/search`,
+        {
+          header: {
+            Accept: "application/json"
+          },
+          params: {
+            api_key: privateApiKey,
+            [id]: userIdentifiers[id]
+          }
+        }
+      );
+      if (profileResponse.success && profileResponse.response?.data?.id) {
+        return profileResponse.response.data.id;
+      }
+    }
+  }
+  return false;
+};
+
+// A sigle func to handle the addition of user to a list
+// from an identify call.
+// DOCS: https://www.klaviyo.com/docs/api/v2/lists
+
+const addUserToList = (message, traitsInfo, conf, destination) => {
+  // Check if list Id is present in message properties, if yes override
+  let targetUrl = `${BASE_ENDPOINT}/api/v2/list/${destination.Config.listId}`;
+  if (get(traitsInfo.properties, "listId")) {
+    targetUrl = `${BASE_ENDPOINT}/api/v2/list/${get(
+      traitsInfo.properties,
+      "listId"
+    )}`;
+  }
```

Comment:
Despite using get to find the same values, we can use optional chaining and store the data in one variable.

Reply:
updated